grails-app/services/au/org/ala/ecodata/CommentService.groovy

```diff
@@ -94,9 +100,16 @@ class CommentService {
      */
     @Transactional
     Comment update(Object json){
+        Boolean update = false
         Comment comment = Comment.get(json.id);
         if (comment) {
             if (comment.userId == json.userId) {
+                update = true;
+            }  else if(doesUserHavePrivilege(json.userId, json.entityId, json.entityType) || json.isALAAdmin){
+                update = true;
+            }
+
+            if(update){
                 comment.text = json.text;
                 //update time
                 comment.dateCreated = new Date();
```

Comment:
Is dateCreated the right value to edit?  If you add a lastUpdated field to the Domain object it should be updated by GORM.